client/checkpoint_delete_test.go

```diff
@@ -24,6 +25,9 @@ func TestCheckpointDeleteError(t *testing.T) {
 	if err == nil || err.Error() != "Error response from daemon: Server error" {
 		t.Fatalf("expected a Server Error, got %v", err)
 	}
+	if !errdefs.IsSystem(err) {
```

Comment:
Not sure I understand these checks here. We have a mock client and are testing that the mock does what we just told it to do?
Seems strange... of course I know you didn't write this.